website/other/bezier-rs-demos/wasm/src/subpath.rs

```diff
@@ -154,4 +154,19 @@ impl WasmSubpath {
 
 		wrap_svg_tag(format!("{subpath_svg}{line_svg}{intersections_svg}"))
 	}
+
+	pub fn self_intersections(&self) -> String {
+		let subpath_svg = self.to_default_svg();
+		let self_intersections_svg = self
+			.0
+			.self_intersections(None, None)
+			.iter()
+			.map(|intersection_t| {
+				let point = &self.0.evaluate(ComputeType::Parametric(*intersection_t));
+				draw_circle(*point, 4., RED, 1.5, WHITE)
```

Comment:
Lots of reference shenanigans going on here. Could you verify that all the `*`s and `&`s are necessary in this lambda?